logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,180 @@
+
+  private final String mBaseLogsDir;
+  private int mPort;
+  private ServerSocket mServerSocket;
+  private final int mMinNumberOfThreads;
+  private final int mMaxNumberOfThreads;
+  private ExecutorService mThreadPool;
+  private volatile boolean mStopped;
+
+  /**
+   * Construct an {@link AlluxioLogServerProcess} instance.
+   *
+   * @param baseLogsDir base directory to store the logs pushed from remote Alluxio servers
+   */
+  public AlluxioLogServerProcess(String baseLogsDir) {
+    mPort = Configuration.getInt(PropertyKey.LOGSERVER_PORT);
+    // The log server serves the logging requests from Alluxio servers.
+    mMinNumberOfThreads = Configuration.getInt(PropertyKey.LOGSERVER_THREADS_MIN);
+    mMaxNumberOfThreads = Configuration.getInt(PropertyKey.LOGSERVER_THREADS_MAX);
+    mBaseLogsDir = baseLogsDir;
+    mStopped = true;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * Create and start logging server and client thread pool.
+   */
+  @Override
+  public void start() throws Exception {
+    SynchronousQueue<Runnable> synchronousQueue = new SynchronousQueue<>();
+    mThreadPool =
+        new ThreadPoolExecutor(mMinNumberOfThreads, mMaxNumberOfThreads,
+            THREAD_KEEP_ALIVE_TIME_MS, TimeUnit.MILLISECONDS, synchronousQueue);
+    mStopped = false;
+    try {
+      mServerSocket = new ServerSocket(mPort);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to bind to port {}.", e);
+    }
+    while (!mStopped) {
+      Socket client;
+      try {
+        client = mServerSocket.accept();
+      } catch (IOException e) {
+        try {
+          ServerSocket tmpServerSocket = new ServerSocket(mPort);
+          // If the code reaches here, it indicates mServerSocket has been closed.
+          // Therefore, by assigning the reference, we have successfully re-binded
+          // mServerSocket.
+          mServerSocket = tmpServerSocket;
+        } catch (IOException e1) {
+          // The original mServerSocket is not closed, therefore just ignore the exception
+        }
+        continue;
+      }
+      InetAddress inetAddress = client.getInetAddress();
+      AlluxioLog4jSocketNode clientSocketNode = new AlluxioLog4jSocketNode(this, client);
+      while (true) {
+        try {
+          mThreadPool.execute(clientSocketNode);
+          break;
+        } catch (RejectedExecutionException e) {
+          // Alluxio log clients (master, secondary master, proxy and workers establish
+          // long-living connections with the log server. Therefore, if the log server cannot
+          // find a thread to service a log client, it is very likely due to low number of
+          // worker threads. If retry fails, then it makes sense just to let system throw
+          // an exception. The system admin should increase the thread pool size.
+          throw new RuntimeException(
+              "Increase the number of worker threads in the thread pool", e);
```

Comment:
In the detail message, first describe what the exception means, then recommend a way to fix it. It would be good to include the specific configuration property that the user needs to increase

Reply:
Done.